On macos 10.15:

```
builder for '/nix/store/zwh6gl89m5il2jrld0ih6al9ylbva1sf-arrow-cpp-8.0.0.drv' failed with exit code 1; last 10 log lines:
  [734/738] Generating helloworld.pb.cc, helloworld.pb.h, helloworld.grpc.pb.cc, helloworld.grpc.pb.h
  [735/738] Building CXX object examples/arrow/CMakeFiles/flight-grpc-example.dir/flight_grpc_example.cc.o
  [736/738] Building CXX object examples/arrow/CMakeFiles/flight-grpc-example.dir/helloworld.pb.cc.o
  [737/738] Building CXX object examples/arrow/CMakeFiles/flight-grpc-example.dir/helloworld.grpc.pb.cc.o
  [738/738] Linking CXX executable release/flight-grpc-example
  FAILED: release/flight-grpc-example
  : && /nix/store/63ds6x8smvmy6q3y8870awxnx5yhwkpp-clang-wrapper-11.1.0/bin/clang++ -Qunused-arguments -fcolor-diagnostics -O3 -DNDEBUG  -Wall -Wno-unknown-warning-option -Wno-pass-failed -msse4.2  -O3 -DNDEBUG -Wl,-search_paths_first -Wl,-headerpad_max_install_names  examples/arrow/CMakeFiles/flight-grpc-example.dir/flight_grpc_example.cc.o examples/arrow/CMakeFiles/flight-grpc-example.dir/helloworld.pb.cc.o examples/arrow/CMakeFiles/flight-grpc-example.dir/helloworld.grpc.pb.cc.o -o release/flight-grpc-example  -Wl,-rpath,/tmp/nix-build-arrow-cpp-8.0.0.drv-1/apache-arrow-8.0.0/cpp/build/release -Wl,-rpath,/nix/store/jmagw2c74jhg25ly45kg7gznsvivq6ki-swift-corefoundation-unstable-2018-09-14/Library/Frameworks  release/libarrow_flight.800.0.0.dylib  -Wl,--no-as-needed  /nix/store/a8gjwjclfvc6yidk5ndmwmc69xfcvv60-grpc-1.45.2/lib/libgrpc++_reflection.1.45.2.dylib  -Wl,--as-needed  /nix/store/y4m7xim96b3gfdxl8qxb11lc6kjiyyg3-protobuf-3.19.3/lib/libprotobuf.dylib  /nix/store/cn73pn62vzj11ws696ff8bq7d527jqn4-gflags-2.2.2/lib/libgflags.2.2.2.dylib  release/libarrow.800.0.0.dylib  /nix/store/snp18j7m97ik0fbqnn2vcr68dcq1s05g-brotli-1.0.9-lib/lib/libbrotlienc.dylib  /nix/store/snp18j7m97ik0fbqnn2vcr68dcq1s05g-brotli-1.0.9-lib/lib/libbrotlidec.dylib  /nix/store/snp18j7m97ik0fbqnn2vcr68dcq1s05g-brotli-1.0.9-lib/lib/libbrotlicommon.dylib  /nix/store/kkycn2z7slxlcgad6rs80ih4q45l680z-glog-0.6.0/lib/libglog.0.6.0.dylib  /nix/store/cn73pn62vzj11ws696ff8bq7d527jqn4-gflags-2.2.2/lib/libgflags.2.2.2.dylib  /nix/store/x5cs237xdcrvciq2jz203pzalc4rana5-aws-sdk-cpp-1.9.238/lib/libaws-cpp-sdk-config.dylib  /nix/store/x5cs237xdcrvciq2jz203pzalc4rana5-aws-sdk-cpp-1.9.238/lib/libaws-cpp-sdk-transfer.dylib  /nix/store/x5cs237xdcrvciq2jz203pzalc4rana5-aws-sdk-cpp-1.9.238/lib/libaws-cpp-sdk-identity-management.dylib  /nix/store/x5cs237xdcrvciq2jz203pzalc4rana5-aws-sdk-cpp-1.9.238/lib/libaws-cpp-sdk-cognito-identity.dylib  /nix/store/x5cs237xdcrvciq2jz203pzalc4rana5-aws-sdk-cpp-1.9.238/lib/libaws-cpp-sdk-sts.dylib  /nix/store/x5cs237xdcrvciq2jz203pzalc4rana5-aws-sdk-cpp-1.9.238/lib/libaws-cpp-sdk-s3.dylib  /nix/store/x5cs237xdcrvciq2jz203pzalc4rana5-aws-sdk-cpp-1.9.238/lib/libaws-cpp-sdk-core.dylib  /nix/store/nqapr51bbh419byhdyxaxhx8ggiy5cj2-aws-crt-cpp-0.17.28/lib/libaws-crt-cpp.dylib  /nix/store/caxw9akbvamv7y2cws2zzy8347zgyx1y-aws-c-mqtt-0.7.10/lib/libaws-c-mqtt.1.0.0.dylib  /nix/store/p5j4yll685j699k7jwp3m5l07mjn4iyb-aws-c-event-stream-0.2.7/lib/libaws-c-event-stream.1.0.0.dylib  /nix/store/ka9lkgvjw67a1a299srdas4fcr3hklg2-aws-c-s3-0.1.39/lib/libaws-c-s3.1.0.0.dylib  /nix/store/337j3px4hk8caqmdlzp9saj8y7dvylsx-aws-c-auth-0.6.13/lib/libaws-c-auth.1.0.0.dylib  /nix/store/ng4lkkr1gmx8pz37plpm2zy225l1knn2-aws-c-http-0.6.15/lib/libaws-c-http.1.0.0.dylib  /nix/store/lrqn3nzkjfn975ylib4wivs79a50myxq-aws-c-io-0.11.0/lib/libaws-c-io.1.0.0.dylib  /nix/store/pxji7ja2316rz1j4115dnamj0k4kw2ci-aws-c-compression-0.2.14/lib/libaws-c-compression.1.0.0.dylib  /nix/store/hy3f5m486cz6akdplzvf1wywssrhpax1-aws-c-cal-0.5.17/lib/libaws-c-cal.1.0.0.dylib  -framework Security  /nix/store/z5fwavmy21nv94fdzkvr0lh6h120p0nq-aws-c-sdkutils-0.1.2/lib/libaws-c-sdkutils.1.0.0.dylib  /nix/store/wd1ms8laqbsp3q76lb9awf4wdyjdx3xp-aws-checksums-0.1.12/lib/libaws-checksums.1.0.0.dylib  /nix/store/binmah3kf8x60igzyz1y53bjyk61y59d-aws-c-common-0.7.0/lib/libaws-c-common.1.0.0.dylib  -pthread  -lpthread  /nix/store/6hwp03ypypis09xhxgykwa7y3d6b0805-utf8proc-2.7.0/lib/libutf8proc.dylib  jemalloc_ep-prefix/src/jemalloc_ep/dist//lib/libjemalloc_pic.a  mimalloc_ep/src/mimalloc_ep/lib/mimalloc-1.7/libmimalloc.a  /nix/store/dbcg3vmbgvxd82r7cj6afbd9pkmbwxay-snappy-1.1.9/lib/libsnappy.1.1.9.dylib  /nix/store/rw1sy2cn0n8vc52agyc44xr7mmwzay39-zstd-1.5.2/lib/libzstd.1.5.2.dylib  /nix/store/a8gjwjclfvc6yidk5ndmwmc69xfcvv60-grpc-1.45.2/lib/libgrpc++.1.45.2.dylib  /nix/store/a8gjwjclfvc6yidk5ndmwmc69xfcvv60-grpc-1.45.2/lib/libgrpc.23.0.0.dylib  /nix/store/1lph5ls57lmxqvx9vac0aq1ly02kccc6-openssl-1.1.1n/lib/libssl.dylib  /nix/store/1lph5ls57lmxqvx9vac0aq1ly02kccc6-openssl-1.1.1n/lib/libcrypto.dylib  /nix/store/4dbpvb55kfly42a6jy5csim2lp4w3h78-re2-2022-04-01/lib/libre2.dylib  /nix/store/wbx9cv1z0l0f8nghgvr4kzw2ib7y79r6-zlib-1.2.12/lib/libz.dylib  /nix/store/xy7rvjpf4d54f74dplkcrj3y97j85pzh-c-ares-1.18.1/lib/libcares.dylib  /nix/store/a8gjwjclfvc6yidk5ndmwmc69xfcvv60-grpc-1.45.2/lib/libaddress_sorting.23.0.0.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_raw_hash_set.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_hashtablez_sampler.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_exponential_biased.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_hash.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_city.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_wyhash.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_statusor.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_bad_variant_access.2103.0.1.dylib  /nix/store/a8gjwjclfvc6yidk5ndmwmc69xfcvv60-grpc-1.45.2/lib/libgpr.23.0.0.dylib  /nix/store/a8gjwjclfvc6yidk5ndmwmc69xfcvv60-grpc-1.45.2/lib/libupb.23.0.0.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_status.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_random_distributions.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_random_seed_sequences.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_random_internal_pool_urbg.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_random_internal_randen.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_random_internal_randen_hwaes.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_random_internal_randen_hwaes_impl.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_random_internal_randen_slow.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_random_internal_platform.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_random_internal_seed_material.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_random_seed_gen_exception.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_cord.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_bad_optional_access.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_str_format_internal.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_synchronization.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_stacktrace.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_symbolize.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_debugging_internal.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_demangle_internal.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_graphcycles_internal.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_malloc_internal.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_time.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_strings.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_throw_delegate.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_int128.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_strings_internal.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_base.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_raw_logging_internal.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_log_severity.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_spinlock_wait.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_civil_time.2103.0.1.dylib  /nix/store/7jg25hkqbc8mnldwzr8h0p29hjm2m582-abseil-cpp-20210324.2/lib/libabsl_time_zone.2103.0.1.dylib  -framework  CoreFoundation  -framework CoreFoundation  /nix/store/y4m7xim96b3gfdxl8qxb11lc6kjiyyg3-protobuf-3.19.3/lib/libprotobuf.dylib  -lm && :
  ld: unknown option: --no-as-needed
  clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
  ninja: build stopped: subcommand failed.
```

--- a/pkgs/development/libraries/arrow-cpp/default.nix
+++ b/pkgs/development/libraries/arrow-cpp/default.nix
@@ -19,6 +19,7 @@
 , grpc
 , gtest
 , jemalloc
+, libbacktrace
 , lz4
 , minio
 , ninja
@@ -69,21 +70,20 @@ let
 in
 stdenv.mkDerivation rec {
   pname = "arrow-cpp";
-  version = "7.0.0";
+  version = "8.0.0";
 
   src = fetchurl {
-    url =
-      "mirror://apache/arrow/arrow-${version}/apache-arrow-${version}.tar.gz";
-    hash = "sha256-6PSbFJoV7O9OQPz6sbh8ETxrHuGGAFwWnlzfldMamd4=";
+    url = "mirror://apache/arrow/arrow-${version}/apache-arrow-${version}.tar.gz";
+    hash = "sha256-rZoFcFEXyYnBFrrprHBJL+AVBQ4bgPsOOP3ktdhjqqM=";
   };
   sourceRoot = "apache-arrow-${version}/cpp";
 
   ${if enableJemalloc then "ARROW_JEMALLOC_URL" else null} = jemalloc.src;
 
+  # versions are all taken from
+  # https://github.com/apache/arrow/blob/apache-arrow-8.0.0/cpp/thirdparty/versions.txt
+
   ARROW_MIMALLOC_URL = fetchFromGitHub {
-    # From
-    # ./cpp/cmake_modules/ThirdpartyToolchain.cmake
-    # ./cpp/thirdparty/versions.txt
     owner = "microsoft";
     repo = "mimalloc";
     rev = "v1.7.3";
@@ -93,8 +93,15 @@ stdenv.mkDerivation rec {
   ARROW_XSIMD_URL = fetchFromGitHub {
     owner = "xtensor-stack";
     repo = "xsimd";
-    rev = "aeec9c872c8b475dedd7781336710f2dd2666cb2";
-    hash = "sha256-vWKdJkieKhaxyAJhijXUmD7NmNvMWd79PskQojulA1w=";
+    rev = "7d1778c3b38d63db7cec7145d939f40bc5d859d1";
+    hash = "sha256-89AysBUVnTdWyMPazeJegnQ6WEH90Ns7qQInZLMSXY4=";
+  };
+
+  ARROW_SUBSTRAIT_URL = fetchFromGitHub {
+    owner = "substrait-io";
+    repo = "substrait";
+    rev = "e1b4c04a1b518912f4c4065b16a1b2c0ac8e14cf";
+    hash = "sha256-56FSjDngsROSHLjMv+OYAIYqphEu3GzgIMHbgh/ZQw0=";
   };
 
   patches = [
@@ -115,7 +122,10 @@ stdenv.mkDerivation rec {
     gflags
     glog
     gtest
+    libbacktrace
     lz4
+    nlohmann_json # alternative JSON parser to rapidjson
+    protobuf # substrait requires protobuf
     rapidjson
     re2
     snappy
@@ -150,6 +160,9 @@ stdenv.mkDerivation rec {
     "-DARROW_BUILD_SHARED=${if enableShared then "ON" else "OFF"}"
     "-DARROW_BUILD_STATIC=${if enableShared then "OFF" else "ON"}"
     "-DARROW_BUILD_TESTS=ON"
+    "-DARROW_BUILD_INTEGRATION=ON"
+    "-DARROW_BUILD_UTILITIES=ON"
+    "-DARROW_EXTRA_ERROR_CONTEXT=ON"
     "-DARROW_VERBOSE_THIRDPARTY_BUILD=ON"
     "-DARROW_DEPENDENCY_SOURCE=SYSTEM"
     "-DThrift_SOURCE=AUTO" # search for Thrift using pkg-config (ThriftConfig.cmake requires OpenSSL and libevent)
@@ -168,8 +181,10 @@ stdenv.mkDerivation rec {
     # Disable Python for static mode because openblas is currently broken there.
     "-DARROW_PYTHON=${if enableShared then "ON" else "OFF"}"
     "-DARROW_USE_GLOG=ON"
+    "-DARROW_WITH_BACKTRACE=ON"
     "-DARROW_WITH_BROTLI=ON"
     "-DARROW_WITH_LZ4=ON"
+    "-DARROW_WITH_NLOHMANN_JSON=ON"
     "-DARROW_WITH_SNAPPY=ON"
     "-DARROW_WITH_UTF8PROC=ON"
     "-DARROW_WITH_ZLIB=ON"
@@ -177,8 +192,10 @@ stdenv.mkDerivation rec {
     "-DARROW_MIMALLOC=ON"
     # Parquet options:
     "-DARROW_PARQUET=ON"
+    "-DARROW_SUBSTRAIT=ON"
     "-DPARQUET_BUILD_EXECUTABLES=ON"
     "-DARROW_FLIGHT=${if enableFlight then "ON" else "OFF"}"
+    "-DARROW_FLIGHT_TESTING=${if enableFlight then "ON" else "OFF"}"
     "-DARROW_S3=${if enableS3 then "ON" else "OFF"}"
     "-DARROW_GCS=${if enableGcs then "ON" else "OFF"}"
   ] ++ lib.optionals (!enableShared) [

--- a/pkgs/development/python-modules/db-dtypes/default.nix
+++ b/pkgs/development/python-modules/db-dtypes/default.nix
@@ -1,6 +1,7 @@
 { lib
 , buildPythonPackage
-, fetchPypi
+, fetchpatch
+, fetchFromGitHub
 , numpy
 , packaging
 , pandas
@@ -12,10 +13,19 @@ buildPythonPackage rec {
   pname = "db-dtypes";
   version = "1.0.0";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "3070d1a8d86ff0b5d9b16f15c5fab9c18893c6b3d5723cd95ee397b169049454";
+  src = fetchFromGitHub {
+    owner = "googleapis";
+    repo = "python-db-dtypes-pandas";
+    rev = "v${version}";
+    hash = "sha256-7u/E0ICiz7LQfuplm/mkGlWrgGEPqeMwM3CUhfH6868=";
   };
+
+  patches = [
+    (fetchpatch {
+      url = "https://github.com/googleapis/python-db-dtypes-pandas/commit/fb30adfd427d3df9919df00b096210ba1eb1b91d.patch";
+      sha256 = "sha256-39kZtYGbn3U1WXiDTczki5EM6SjUlSRXz8UMcdTU20g=";
+    })
+  ];
 
   propagatedBuildInputs = [
     numpy

--- a/pkgs/development/python-modules/pyarrow/default.nix
+++ b/pkgs/development/python-modules/pyarrow/default.nix
@@ -47,8 +47,10 @@ buildPythonPackage rec {
 
   PYARROW_WITH_DATASET = zero_or_one true;
   PYARROW_WITH_FLIGHT = zero_or_one _arrow-cpp.enableFlight;
-  PYARROW_WITH_PARQUET = zero_or_one true;
   PYARROW_WITH_HDFS = zero_or_one true;
+  PYARROW_WITH_PARQUET = zero_or_one true;
+  PYARROW_WITH_PLASMA = zero_or_one (!stdenv.isDarwin);
+  PYARROW_WITH_S3 = zero_or_one _arrow-cpp.enableS3;
 
   PYARROW_CMAKE_OPTIONS = [
     "-DCMAKE_INSTALL_RPATH=${ARROW_HOME}/lib"
@@ -73,6 +75,11 @@ buildPythonPackage rec {
     # enabled in nixpkgs.
     # Upstream Issue: https://issues.apache.org/jira/browse/ARROW-11393
     "--deselect=pyarrow/tests/test_memory.py::test_env_var"
+    # these tests require access to s3 via the internet
+    "--deselect=pyarrow/tests/test_fs.py::test_resolve_s3_region"
+    "--deselect=pyarrow/tests/test_fs.py::test_s3_real_aws"
+    "--deselect=pyarrow/tests/test_fs.py::test_s3_real_aws_region_selection"
+    "--deselect=pyarrow/tests/test_fs.py::test_s3_options"
   ] ++ lib.optionals stdenv.isDarwin [
     # Requires loopback networking
     "--deselect=pyarrow/tests/test_ipc.py::test_socket_"
@@ -84,16 +91,17 @@ buildPythonPackage rec {
     rm -r pyarrow/!(tests)
   '';
 
-  pythonImportsCheck = map (module: "pyarrow.${module}") [
+  pythonImportsCheck = [ "pyarrow" ] ++ map (module: "pyarrow.${module}") ([
     "compute"
     "csv"
     "dataset"
+    "feather"
     "flight"
     "fs"
     "hdfs"
     "json"
     "parquet"
-  ];
+  ] ++ lib.optionals (!stdenv.isDarwin) [ "plasma" ]);
 
   meta = with lib; {
     description = "A cross-language development platform for in-memory data";

